### PR TITLE
[REF] extract calculation of basic stats

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Code clean up preliminary to resolving performance issues

Before
----------------------------------------
Code hard to read

After
----------------------------------------
Code more readable

Technical Details
----------------------------------------
Partial of https://github.com/civicrm/civicrm-core/pull/13607 - adding 'has-tests' which were merged in #13605

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Contact/BAO/Query.php [hunk 1]
```diff
@@ -5089,44 +5089,27 @@ public function setSkipPermission($val) {
...
-    }
-    if ($context == 'search') {
-      $where .= " AND contact_a.is_deleted = 0 ";
-    }
 
-    $query = $this->appendFinancialTypeWhereAndFromToQueryStrings($where, $from);
+    $this->appendFinancialTypeWhereAndFromToQueryStrings($where, $from);
 
-    // make sure contribution is completed - CRM-4989
-    $completedWhere = $where . " AND civicrm_contribution.contribution_status_id = 1 ";
```
Comment (eileenmcnaughton) [inline CRM/Contact/BAO/Query.php]: I've been moving this clause into the relevant queries - there just isn't enough gained by sharing bits of queries & it makes it more complicated

File: CRM/Contact/BAO/Query.php [hunk 2]
```diff
@@ -5139,6 +5122,8 @@ public function summaryContribution($context = NULL) {
 
     $summary['total']['mode'] = CRM_Contribute_BAO_Contribution::computeStats('mode', $modeSQL);
 
+    // make sure contribution is completed - CRM-4989
+    $completedWhere = $where . " AND civicrm_contribution.contribution_status_id = 1 ";
```
Comment (eileenmcnaughton) [inline CRM/Contact/BAO/Query.php]: for now - but this variable's days are numbered.....

**Comment by colemanw** [comment]
Merging based on code review and passing tests.

**Comment by eileenmcnaughton** [comment]
thanks @colemanw

**Pull request merged**